cluster-autoscaler/cloudprovider/aws/aws_util_test.go

```diff
@@ -136,3 +138,99 @@ func TestGetCurrentAwsRegionWithRegionEnv(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, region, result)
 }
+
+func TestUnmarshalProductsResponse(t *testing.T) {
+	body := `
```

Comment:
Can we add a test for an invalid json response as well as valid?

Reply:
I'll see what I can knock up

Reply:
I've added some tests for invalid json (and fixed a bug in the process). Shout if you'd like me to rebase them down.